.github/workflows/actions.yml

```diff
@@ -117,23 +117,15 @@ jobs:
       - uses: actions/checkout@v2
         with:
           submodules: recursive
-      - name: set sc version and artifact filename
-        run: |
-          # set SC_VERSION to tag (if present) or to commit SHA
-          SC_VERSION=
-          if [[ "$GITHUB_REF" == refs/tags/* ]]; then 
-            FULL_TAG=${GITHUB_REF#refs/tags/}
-            TAG=${FULL_TAG##Version-}
-            SC_VERSION=$TAG
-          else
-            SC_VERSION=$GITHUB_SHA
-          fi
-          echo "ARTIFACT_FILE=SuperCollider-$SC_VERSION-${{ matrix.artifact-suffix }}.zip" >> $GITHUB_ENV
+      - name: prepare daily timestamp for cache
+        id: current-date
+        run: echo "::set-output name=stamp::$(date '+%Y-%m-%d')"
       - name: cache ccache
         uses: actions/cache@v2
         with:
           path: ~/.ccache
-          key: ${{ runner.os }}-v1-${{ matrix.os-version }}-${{ matrix.c-compiler }}-${{ matrix.use-syslibs }}-${{ matrix.shared-libscsynth }} # -v2- etc can be used for "clearing" the cache, it's not functionally connected to the version of the script
+          key: ${{ runner.os }}-${{ matrix.os-version }}-${{ matrix.c-compiler }}-${{ matrix.use-syslibs }}-${{ matrix.shared-libscsynth }}-${{ steps.current-date.outputs.stamp }}
+          restore-keys: ${{ runner.os }}-${{ matrix.os-version }}-${{ matrix.c-compiler }}-${{ matrix.use-syslibs }}-${{ matrix.shared-libscsynth }}-
```

Comment:
what does this line do? is it like a fallback?

Reply:
the way tis works is that `key` is searched first, ending in current `year-month-day`. If that is not found (i.e. the cache is checked the after midnight from the previous run, so current date and cache date don't match), `restore-keys` are searched and the latest one is loaded, in this case cache from previous day.

Reply:
oh i see, we can use it as a pattern to match against other keys

> If there are multiple partial matches for a restore key, the action returns the most recently created cache.

https://docs.github.com/en/free-pro-team@latest/actions/guides/caching-dependencies-to-speed-up-workflows#matching-a-cache-key